src/components/AddCartButton/AddCartButton.component.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  addToCart,
+  calculateTotalCart,
+  calculateTotalProducts,
+} from '../../actions/cart';
+import { useForm } from '../../utils/hooks/useForm/useForm';
+import {
+  AddButton,
+  AddCartContainer,
+  NumberProducts,
+} from './AddCartButton.style';
+const AddCartButton = () => {
+  const dispatch = useDispatch();
+  const { product } = useSelector((state) => state.cart);
+  const {
+    product: { stock },
```

Comment:
This not looks correct since you are selecting `(state) => state.cart` and it is returning `product`, it should be `products`, right?